torchtitan/parallelisms/pipelining_utils.py

```diff
@@ -26,7 +33,7 @@ def build_pipeline_schedule(job_config, parallel_dims, stages, loss_fn):
         n_microbatches = job_config.experimental.pipeline_parallel_degree
 
     return schedule_class(
-        stage,
+        stages if looped_schedule else stages[0],
```

Comment:
the init function of ScheduleInterleaved1F1B takes a list of _PipelineStageBase
if we only pass stages[0], will it cause ScheduleInterleaved1F1B fail ?

Reply:
well, the point of this code is that the schedule class could be a simple schedule or a looped schedule.

If it's a looped schedule, like Interleaved1F1B, then we must have 'stages'.

if its a simple schedule, then we must have just one stage, so 'stages[0]' is appropriate.

But we should never have Interleaved1F1B(stages[0]).